internal/healthchecks/error.go

```diff
@@ -0,0 +1,129 @@
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package healthchecks
+
+// Error classification
+const (
+	Api        = "API"
+	Connection = "CONNECTION"
+	Generic    = "GENERIC"
+	Port       = "PORT"
+	Permission = "PERMISSION"
+)
+
+type HealthCheckError struct {
+	Code         string
+	Class        string
+	Message      string
+	Action       string
+	ResourceLink string
+	IsFatal      bool
+	Err          error
+}
+
+func (e HealthCheckError) Error() string {
+	return e.Err.Error()
+}
+
+var (
+	FbMetricsPortErr = HealthCheckError{
+		Code:         "FbMetricsPortErr",
+		Class:        Port,
+		Message:      "Port 20202 needed for fluent-bit self metrics is unavailable.",
+		Action:       "Verify that port 20202 is open.",
+		ResourceLink: "https://cloud.google.com/logging/docs/agent/ops-agent/troubleshooting",
+		IsFatal:      true,
+	}
+	OtelMetricsPortErr = HealthCheckError{
+		Code:         "OtelMetricsPortErr",
+		Class:        Port,
+		Message:      "Port 20201 needed for OpenTelemetry self metrics is unavailable.",
+		Action:       "Verify that port 20201 is open.",
+		ResourceLink: "https://cloud.google.com/monitoring/agent/ops-agent/troubleshooting",
+		IsFatal:      true,
+	}
+	LogApiConnErr = HealthCheckError{
+		Code:         "LogApiConnErr",
+		Class:        Connection,
+		Message:      "Request to Logging API failed.",
+		Action:       "Check your internet connection and firewall rules.",
+		ResourceLink: "https://cloud.google.com/logging/docs/agent/ops-agent/troubleshooting",
+		IsFatal:      true,
+	}
+	MonApiConnErr = HealthCheckError{
+		Code:         "MonApiConnErr",
+		Class:        Connection,
+		Message:      "Request to Monitoring API failed.",
+		Action:       "Check your internet connection and firewall rules.",
+		ResourceLink: "https://cloud.google.com/monitoring/agent/ops-agent/troubleshooting",
+		IsFatal:      true,
+	}
+	LogApiScopeErr = HealthCheckError{
+		Code:         "LogApiScopeErr",
+		Class:        Permission,
+		Message:      "VM is missing the https://www.googleapis.com/auth/logging.write scope.",
+		Action:       "Add the https://www.googleapis.com/auth/logging.write scope to the Google Compute Engine VM.",
+		ResourceLink: "https://cloud.google.com/stackdriver/docs/solutions/agents/ops-agent/authorization",
+		IsFatal:      true,
+	}
+	MonApiScopeErr = HealthCheckError{
+		Code:         "MonApiScopeErr",
+		Class:        Permission,
+		Message:      "VM is missing the https://www.googleapis.com/auth/monitoring.write scope.",
+		Action:       "Add the https://www.googleapis.com/auth/monitoring.write scope to the Google Compute Engine VM.",
```

Comment:
the Compute Engine VM (see comment at line 77.

Reply:
Addressed in https://github.com/GoogleCloudPlatform/ops-agent/pull/1051/commits/17c9e03b8b5c8495b0b10e8de8417db0cd7b0510 .